src/ruby/puppet-server-lib/puppet/server/oids.rb

```diff
@@ -0,0 +1,31 @@
+module Puppet::Server::Oids
+  def self.load_custom_oid_file(custom_oid_file, map_key='oid_mapping')
+    if File.exists?(custom_oid_file) && File.readable?(custom_oid_file)
```

Comment:
Are there any cases where a File will be readable but not exist? Would the `readable?` check on its own be sufficient here?